bluesky/callbacks/mpl_plotting.py

```diff
@@ -7,14 +7,242 @@
 from event_model import DocumentRouter
 
 
+class RunFigureManager:
+    def __init__(self):
+
+
 class FigureManager:
     def __init__(self):
+        self.noplot_streams = []  # Do we want to keep this?
         ...
 
     def __call__(self, name, doc):
         if name == 'start':
+            self._start_doc = doc
+            self.plan_hints = doc.get('hints', {})
+
+            # Prepare a guess about the dimensions (independent variables) in case
+            # we need it.
+            motors = self._start_doc.get('motors')
+            if motors is not None:
+                GUESS = [([motor], 'primary') for motor in motors]
+            else:
+                GUESS = [(['time'], 'primary')]
+
+            # Ues the guess if there is not hint about dimensions.
+            dimensions = self.plan_hints.get('dimensions')
+            if dimensions is None:
+                self._cleanup_motor_heuristic = True
+                dimensions = GUESS
+
+            # We can only cope with all the dimensions belonging to the same
+            # stream unless we resample. We are not doing to handle that yet.
+            if len(set(d[1] for d in dimensions)) != 1:
+                self._cleanup_motor_heuristic = True
+                dimensions = GUESS  # Fall back on our GUESS.
+                warn("We are ignoring the dimensions hinted because we cannot "
+                    "combine streams.")
+
+            # for each dimension, choose one field only
+            # the plan can supply a list of fields. It's assumed the first
+            # of the list is always the one plotted against
+            self.dim_fields = [fields[0]
+                            for fields, stream_name in dimensions]
+
+            # make distinction between flattened fields and plotted fields
+            # motivation for this is that when plotting, we find dependent variable
+            # by finding elements that are not independent variables
+            self.all_dim_fields = [field
+                                for fields, stream_name in dimensions
+                                for field in fields]
+
+            _, self.dim_stream = dimensions[0]
             return self
-        # Decide what plot(s) we need for this stream.
+
+        # Decide which plots we want for this stream.
+
+        self._descriptors[doc['uid']] = doc
+        stream_name = doc.get('name', 'primary')  # fall back for old docs
+        columns = hinted_fields(doc)
```

Comment:
Should we filter this by things that are plot-able?

Reply:
We check `ndim` further down. Is that what you mean?

Reply:
strings aren't plottable